File: Zestawy Zadań/C++/5 - Funkcje, rekurencja/Zestaw 2 zadania/5.18 Funkcja o prototypie string ZZ(string S, char TZ[], int n).cpp
#include <iostream>

double S4(int n) {
    if (n == 1) {
        return 1;
    } else {
        return 1.0 / (n * (2 * n - 1)) + S4(n - 1);
    }
}

int main() {
    int N;
    std::cout << "Podaj liczbê ca³kowit¹: ";
    std::cin >> N;
    std::cout << "Wynik: " << S4(N) << std::endl;
    return 0;
}

File: Zestawy Zadań/Python/5 - Funkcje, rekurencja/Zestaw 1 zadania/5.18 Funkcja o prototypie string ZZ(string S, char TZ[], int n).py
def S4(n):
    if n == 1:
        return 1
    else:
        return 1 / (n * (2 * n - 1)) + S4(n - 1)

if __name__ == "__main__":
    N = int(input("Podaj liczbę całkowitą: "))
    print("Wynik:", S4(N))
